### DISCUSSION TITLE
Use _aggregate in deep

### DISCUSSION THREAD

**Discussion by TranQuangMinh** [directus/directus#12925] [Archive]
### Describe the Bug

collections
  - runners (id, name, ...)
  - activities (...distance... owner_id relation with runners.id)
  
 I need listing runner and sum distance by owner_id from  activities,
but response alway list all activities by runners.id

### To Reproduce

 ```js
let runners = await directus.items('runners').readByQuery({
        limit: -1,
        fields: '*.*',
        deep: {
            activities: {
                _aggregate: {
                    _sum: "distance"
                }
            }
        }
    })

```

<details>
<summary>More info</summary>

### What version of Directus are you using?

9.8

### What version of Node.js are you using?

17.4.0

### What database are you using?

mysql

### What browser are you using?

chrome

### How are you deploying Directus?

locally

</details>

**Comment by azrikahar** [comment]
I believe it should be `sum` instead of `_sum` in your current query. Can you try that instead?

**Reply by TranQuangMinh** [reply]
```
let partner = await directus.items('runners').readByQuery({
        //search: keyword,
        limit: -1,
        fields: 'id,activities.*',
        deep: {
            activities: {
                _limit: 1,
                //_sort: '+distance',
                _aggregate: {
                    max: "distance"
                }
            }
        }
})
```

<img width="1011" alt="image" src="https://user-images.githubusercontent.com/10704241/164444774-73eabe38-3717-4e84-a45f-74226ee3d63a.png">

<img width="867" alt="image" src="https://user-images.githubusercontent.com/10704241/164444975-bcdc8559-f0b4-4446-ba4f-ed78ac761931.png">

![Image Pasted at 2022-4-21 18-06](https://user-images.githubusercontent.com/10704241/164445466-c28cecf8-bb4f-430e-8d0d-4d490266ab9f.png)

**Reply by TranQuangMinh** [reply]
if use max instead _max, agg only show last item

**Reply by azrikahar** [reply]
Seems like the types are incorrect indeed 👍 

---

With that said, you can probably try something like:

```js
let runners = await directus.items('runners').readByQuery({
       limit: -1,
       fields: ['*', 'activities.sum'],
       deep: {
           activities: {
               _aggregate: { sum: "distance" },
               _groupBy: ['owner_id'],
               _sort: 'owner_id',
               _limit: -1
           }
       }
   })
```

**Reply by TranQuangMinh** [reply]
yeah, it woking. Thanks so much

**Reply by daodinhduc1998** [reply]
> Seems like the types are incorrect indeed 👍
> 
> With that said, you can probably try something like:
> 
> ```js
> let runners = await directus.items('runners').readByQuery({
>        limit: -1,
>        fields: ['*', 'activities.sum'],
>        deep: {
>            activities: {
>                _aggregate: { sum: "distance" },
>                _groupBy: ['owner_id'],
>                _sort: 'owner_id',
>                _limit: -1
>            }
>        }
>    })
> ```

@azrikahar I'm wanting to sort the records by "sum" field so how should I do it here ?
